scripts/arbitrum/2_2_deploy_beefy_depository.ts

```diff
@@ -0,0 +1,102 @@
+import { ethers, network, upgrades } from "hardhat";
+import {
+    BeefyDepository,
+    UXDController,
+    UXDRouter,
+} from "../../typechain-types";
+import { verify } from "../common/verify";
+import fs from "fs";
+import { loadConfig, loadCoreContracts } from "../common/loaders";
+import { ArbitrumConfig } from "../../config/configs";
+import { CoreContracts } from "../../config/contracts";
+import { checkNetwork, NetworkName } from "../common/checkNetwork";
+
+let controller: UXDController;
+let depository: BeefyDepository;
+let router: UXDRouter;
+let config: ArbitrumConfig;
+let coreContracts: CoreContracts;
+
+async function main() {
+    const hre = await import("hardhat")
+    checkNetwork(hre, [NetworkName.ArbitrumGoerli, NetworkName.ArbitrumOne]);
+
+    config = (await loadConfig(hre)) as ArbitrumConfig;
+    coreContracts = await loadCoreContracts(hre);
+
+    const UXDController = await ethers.getContractFactory("UXDController");
+    controller = UXDController.attach(coreContracts.controller) as UXDController;
+
+    const UXDRouter = await ethers.getContractFactory("UXDRouter");
+    router = UXDRouter.attach(coreContracts.router) as UXDRouter;
+
+    console.log(`controller address is ${controller.address}`)
+    console.log(`vault address is ${config.contracts.BeefyWrapperUSDCVault}`)
+    const BeefyDepository = await ethers.getContractFactory("BeefyDepository");
+
+    depository = await upgrades.deployProxy(BeefyDepository, [
+        config.contracts.BeefyWrapperUSDCVault,
+        controller.address
+    ],
+        { kind: "uups" }
+    ) as BeefyDepository;
+
+    await depository.deployed();
+    console.log(`BeefyDepository deployed to ${depository.address}`);
+
+    await setup();
+    await verifyContracts();
+    save();
+    console.log("Complete 🚀🚀🚀");
+}
+
+async function setup() {
+    console.log(`In Setup router address = ${router.address}, depository=${depository.address}`)
+    console.log(
+        `Registering depository => [asset=${config.tokens.USDC}, depository=${depository.address}]`
+    );
+    await (
+        await router.registerDepository(depository.address, config.tokens.USDC)
+    ).wait();
+
+    const supplyCap = ethers.utils.parseEther("10000000");
+
+    console.log(
+        `Setting depository soft cap => ${ethers.utils.formatEther(supplyCap)}`
+    );
+    await (await depository.setRedeemableSoftCap(supplyCap)).wait();
+}
+
+async function verifyContracts() {
+    console.log(`> Verifying depository: ${depository.address}`)
+    await verify(depository.address);
+
+    const depositoryImpl = await upgrades.erc1967.getImplementationAddress(depository.address)
+    console.log(`> Verifying depositoryImpl: ${depositoryImpl}`)
+    await verify(depositoryImpl);
+}
+
+function save() {
+    console.log("Saving contract addresses...");
+    const config = `
+  {
+    "rageTrade": "${depository.address}"
```

Comment:
We currently have only one depository per chain. If we have multiple we would want to first read the depositories file, then add to that, so in this case you should write:
```
{
    "rageTrade": "${depositories.rageTrade}", // just re-writing the existing value
    "beefyFinance": "${depository.address}" // write the new depository address
}
```
Current script would overwrite the rageTrade depository address with the beefy one.

Reply:
Would also need to load that `depositories.rageTrade` from somewhere first?